### DISCUSSION TITLE
SubjectRule for several Fields

### DISCUSSION THREAD

**Discussion by Yann57330** [Sleepw4lker/TameMyCerts#12] [Q&A] [answered]
Hello,

Thanks for this great Module ! Really usefull
I'm trying to allow to use CN and OU in Subject field.
Your example illustrate only Subect rule for CN.
How can we manage several SubjectRule for several fields ?
I have tried to indicate 2  <SubjectRule> inside the <Subject> with no success.

Thanks for your help.
BR,
Yannick

**Comment by Sleepw4lker** [comment]
Hello @Yann57330 and thanks for your feedback: Have a look at this: https://github.com/Sleepw4lker/TameMyCerts/blob/main/user-guide/subject-rules.adoc and this: https://github.com/Sleepw4lker/TameMyCerts/blob/main/examples/Sample_Offline_Webserver.xml.

You would have to define another SubjectRule directive containing the **organizationalUnitName** field. 

Hope that helps
Cheers

**Reply by Yann57330** [reply]
Hi,

Thanks. I have tried with 2 SubjectRule 

`<Subject>
    <SubjectRule>
        <Field>commonName</Field>
        <Mandatory>true</Mandatory>
        <MaxOccurrences>1</MaxOccurrences>
        <MaxLength>64</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^[a-zA-Z0-9_\-]*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
	<SubjectRule>
        <Field>organizationalUnitName</Field>
        <Mandatory>false</Mandatory>
        <MaxOccurrences>1</MaxOccurrences>
        <MaxLength>100</MaxLength>
        <Patterns>
            <Pattern>
                <Expression>^[a-zA-Z0-9_\.\-\:\[\]]*$</Expression>
            </Pattern>
        </Patterns>
    </SubjectRule>
</Subject>`

That seems not to work.
I will carry on with this syntax to troubleshoot as your example for Webserver confirm its...

BR,
Yannick

**Comment by Sleepw4lker** [comment, marked as answer]
Hello @Yann57330 your configuration should work, given you included the <CertificateRequestPolicy... headers and footers in your XML file. I can only assume that maybe your regular expression does not match. Did you have a look at the event log of your machine? Does the request get denied (then, for whar reason?) or approved (is the policy applied at all)?

**Reply by Yann57330** [reply]
Hello, @Sleepw4lker.
I’ve tried again from start today and it is perfectly working ! :-) 
Thanks a lot for your fast answer yesterday.
You are probably right, there was maybe missing something in the pattern…
I’ve not checked in event log. I will for futur.
Thanks again for the module. Great work.
Kind regards,
Yannick

**Reply by Sleepw4lker** [reply]
any update on this @Yann57330 ?

Cheers

**Reply by Yann57330** [reply]
Yes @Sleepw4lker it works as mentioned in my previous comment.

Kind regards

**Reply by Sleepw4lker** [reply]
Awesome, thanks for the clarification.
